overlord/auth/auth.go

```diff
@@ -440,10 +441,11 @@ func (ac *authContext) UpdateUserAuth(user *UserState, newDischarges []string) (
 	return cur, nil
 }
 
-// StoreID returns the store set in the model assertion, if mod != nil,
-// or the override from the UBUNTU_STORE_ID envvar.
+// StoreID returns the store set in the model assertion, if mod != nil
+// and it's not the generic classic model, or the override from the
+// UBUNTU_STORE_ID envvar.
 func StoreID(mod *asserts.Model) string {
-	if mod != nil {
+	if mod != nil && mod.Ref().Unique() != sysdb.GenericClassicModel().Ref().Unique() {
```

Comment:
h> 09:25 < zyga-ubuntu> mvo: silly question, which store is used on the generic model when UBUNTU_STORE_ID is unset?
> 09:26 < zyga-ubuntu> mvo: should we not return mod.Store() if UBUNTU_STORE_ID is unset?
> 09:27 < mvo> zyga-ubuntu: hm, good point
> 09:27 < zyga-ubuntu> (nothing like silly questions)
> 09:27 < zyga-ubuntu> mvo: let's tweak that test to have a case like that
> 09:27 < zyga-ubuntu> if the outcome is sensible then +1 but I think it may be a missing cae
> 09:27 < zyga-ubuntu> *case
> 09:28 < mvo> zyga-ubuntu: looks like the generic model is not defining a store so the end result is the same, but I agree with you, it seems cleaner to not rely on this
>

Reply:
I added a test case that shows what happens when we are using the generic model but UBUNTU_STORE_ID is not set. I think @pedronis knows better what the expected outcome is so let's wait for his feedback.

Reply:
yes, that's fine, is the same behavior as if there's no model at all (though that doesn't quite happen anymore, except temporarely)